Storage/REST-API.md

```diff
@@ -0,0 +1,183 @@
+# RESTful binding for Storage, RFC
+
+This document describes the RESTful API with CRUD functionality for Current Storage.
+
+### Bindings semantics
+`TODO: Describe storage field <-> endpoint bindings.`
+
+### Response schema
+
+The response body is always either a valid JSON or empty.
+
+The generic format of a successful JSON response is:
+
+```
+{
+  "url": "https://api.example.com/...",
+  "url_*": "https://api.example.com/...",
+  ... more "url_*" fields ...
+  
+  [optional]
+  "meta": {
+    ...
+  },
+  
+  [required]
+  "data": {
+    ...
+  }
+
+}
+```
+
+The responses are strongly typed. The contents of the request URL define the schema for the response.
+
+The value for `"data"` can be:
+* an object,
+* an array, if the response is expected to contain a list,
+* primitive type (string, number, boolean) in special cases.
+
+Error responses are JSON objects with the top-level `"error":...` field:
+
+```
+{
+  “error”: “Detailed human-readable error message.”,
```

Comment:
@mzhurovich  What do you think of making `error` an object with a human-readable `message` field and other error detail fields next to it, like the exception name (code)? Error objects are quite standard, some APIs even respond with arrays of error objects. Otherwise, to reliably determine an error condition from the response, you have to guarantee that the human-readable message is always present and non-empty.